jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceCollection.java

```diff
@@ -130,6 +125,15 @@ public Resource resolve(String subUriPath)
         Resource addedResource = null;
         for (Resource res : _resources)
         {
+            // since we allow non-directory resources, we need a way to resolve them
+            // -> match if the subUriPath contains no / and matches the resource's filename.
+            // TODO review this
+            String subUriStringNotPrependedBySlash = subUriPath;
+            if (subUriStringNotPrependedBySlash.startsWith("/"))
+                subUriStringNotPrependedBySlash = subUriStringNotPrependedBySlash.substring(1);
```

Comment:
This as never been necessary before.  Resource has always treated leading slash and no leading slash the same, so we should not have to do this.... and if we did, it should not be done in the loop.

Reply:
This is a quick & ugly hack needed to demonstrate what it takes to be able to serve non-dir resources added to a ResourceCollection.

In case we think this mechanism is a good idea, I'll come up with a clean implementation to replace this horrible hack.